sources/packages/web/src/views/aest/institution/DesignationAESTView.vue

```diff
@@ -0,0 +1,221 @@
+          class="ml-2 primary-btn-background"
+          v-if="showActionButtons"
+          @click="updateDesignation(DesignationAgreementStatus.Approved)"
+          >Approve designation</v-btn
+        >
+      </template>
+    </header-navigator>
+    <full-page-container class="mt-4">
+      <designation-agreement-form
+        :model="designationFormModel"
+      ></designation-agreement-form>
+    </full-page-container>
+    <approve-deny-designation
+      ref="approveDenyDesignationModal"
+      :designation="updateDesignationModel"
+    />
+  </div>
+</template>
+
+<script lang="ts">
+import FullPageContainer from "@/components/layouts/FullPageContainer.vue";
+import { onMounted, reactive, ref, computed } from "vue";
+import { RouteLocationRaw } from "vue-router";
+import {
+  useFormatters,
+  useDesignationAgreement,
+  ModalDialog,
+  useToastMessage,
+} from "@/composables";
+import DesignationAgreementForm from "@/components/partial-view/DesignationAgreement/DesignationAgreementForm.vue";
+import { DesignationAgreementService } from "@/services/DesignationAgreementService";
+import { InstitutionService } from "@/services/InstitutionService";
+import {
+  UpdateDesignationDto,
+  DesignationAgreementStatus,
+  UpdateDesignationLocationDto,
+  GetDesignationAgreementDto,
+} from "@/types/contracts/DesignationAgreementContract";
+import {
+  DesignationModel,
+  DesignationFormViewModes,
+} from "@/components/partial-view/DesignationAgreement/DesignationAgreementForm.models";
+import HeaderNavigator from "@/components/generic/HeaderNavigator.vue";
+import { AESTRoutesConst } from "@/constants/routes/RouteConstants";
+import ApproveDenyDesignation from "@/views/aest/institution/ApproveDenyDesignation.vue";
+
+export default {
+  components: {
+    HeaderNavigator,
+    DesignationAgreementForm,
+    FullPageContainer,
+    ApproveDenyDesignation,
+  },
+  props: {
+    designationAgreementId: {
+      type: Number,
+      required: true,
+    },
+    institutionId: {
+      type: Number,
+      required: true,
+    },
+  },
+  setup(props: any) {
+    const formatter = useFormatters();
+    const { mapDesignationChipStatus } = useDesignationAgreement();
+    const designationAgreement = ref({} as GetDesignationAgreementDto);
+    const designationFormModel = reactive({} as DesignationModel);
+    const showModal = ref(false);
+    const toast = useToastMessage();
+    const showActionButtons = computed(
+      () =>
+        designationFormModel.designationStatus !==
+        DesignationAgreementStatus.Declined,
+    );
+    const approveDenyDesignationModal = ref(
+      {} as ModalDialog<UpdateDesignationDto | boolean>,
+    );
+    const updateDesignationModel = ref({} as UpdateDesignationDto);
+    let navigationTitle = "Pending designations";
+    let routeLocation: RouteLocationRaw = {
+      name: AESTRoutesConst.PENDING_DESIGNATIONS,
+    };
+    if (props.institutionId) {
+      navigationTitle = "Manage designations";
```

Comment:
I think you can use  `computed ` for `navigationTitle`  and `routeLocation`

Reply:
computed used.